zyte_api/__main__.py

```diff
@@ -23,35 +23,52 @@ class DontRetryErrorsFactory(RetryFactory):
     retry_condition = retry_if_exception(_is_throttling_error)
 
 
-logger = logging.getLogger('zyte_api')
+logger = logging.getLogger("zyte_api")
 
 _UNSET = object()
 
 
-async def run(queries, out, *, n_conn, stop_on_errors, api_url,
-              api_key=None, retry_errors=True):
+def write_output(result, out, pbar):
+    json.dump(result, out, ensure_ascii=False)
+    out.write("\n")
+    out.flush()
+    pbar.update()
+
+
+async def run(
+    queries,
+    out,
+    *,
+    n_conn,
+    stop_on_errors,
+    api_url,
+    api_key=None,
+    retry_errors=True,
+    include_all_responses,
+):
     retrying = None if retry_errors else DontRetryErrorsFactory().build()
-    client = AsyncClient(n_conn=n_conn, api_key=api_key, api_url=api_url,
-                         retrying=retrying)
+    client = AsyncClient(
+        n_conn=n_conn, api_key=api_key, api_url=api_url, retrying=retrying
+    )
     async with create_session(connection_pool_size=n_conn) as session:
         result_iter = client.request_parallel_as_completed(
             queries=queries,
             session=session,
         )
-        pbar = tqdm.tqdm(smoothing=0, leave=True, total=len(queries), miniters=1,
-                         unit="url")
+        pbar = tqdm.tqdm(
+            smoothing=0, leave=True, total=len(queries), miniters=1, unit="url"
+        )
         pbar.set_postfix_str(str(client.agg_stats))
         try:
             for fut in result_iter:
                 try:
                     result = await fut
-                    json.dump(result, out, ensure_ascii=False)
-                    out.write("\n")
-                    out.flush()
-                    pbar.update()
+                    write_output(result, out, pbar)
                 except Exception as e:
-                    if stop_on_errors:
+                    if stop_on_errors or not include_all_responses:
```

Comment:
This seems wrong, raising an exception because the user did not ask for errors to be stored.